src/data-model-providers/codegen/model.gni

```diff
@@ -29,6 +29,8 @@ codegen_data_model_SOURCES = [
   "${chip_root}/src/data-model-providers/codegen/CodegenDataModelProvider.h",
   "${chip_root}/src/data-model-providers/codegen/CodegenDataModelProvider_Read.cpp",
   "${chip_root}/src/data-model-providers/codegen/CodegenDataModelProvider_Write.cpp",
+  "${chip_root}/src/data-model-providers/codegen/CodegenServerCluster.h",
```

Comment:
could we isolate this more? Right now this is fully placed under codegen which is more "final/official implementation" and we would force every platform to compile it and rely on LTO to not affect sizes.

How about having this under some "examples/shim" folder instead?